woocommerce/woocommerce-ios
# [Subscriptions in Order Creation 3] Update order details row with subscription details

Closes: #12441 

## Description
This PR updates the Subscription details row within Order Details when an order has a subscription product, following the figma designs at Q9tkFMtymZz0yhfLLbKhig-fi-2176_61285 (under `i1 - 06 - Order Details`):

<img width="360" alt="Screenshot 2024-04-24 at 16 02 09" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/5a4e83f9-c169-4441-b58c-f160851f6d37">

| Before | After |
|--------|--------|
| ![20240424 order details subscription section - before ](https://github.com/woocommerce/woocommerce-ios/assets/3812076/99fdc06b-47d1-4794-89f4-c27a40ad9762) | ![Simulator Screenshot - iPhone 15 - 2024-04-24 at 15 57 07](https://github.com/woocommerce/woocommerce-ios/assets/3812076/78c07bc2-3fee-4418-a4a1-5d2d2979541d) | 

## Changes
- Reordered order details rows, now `subscriptions` section appears before `payment` section.
- Updated existing `OrderSubscriptionTableViewCell` for the new design. Mostly redo the .xib file and add a new label and a separator.
- Updated cell strings to be formatted via `localizedStringWithFormat`
- Cleanup: Moved `OrderSubscriptionTableViewCellViewModel` to a separate file
- Cleanup: Updated the `subscriptionTitle` to not appear as a decimal number
- Cleanup: Updated affected `NSLocalizedStrings` to use `value:`

## Testing instructions
- On a store using WooCommerce Subscriptions, process an order with subscription products through the web (not the app, since is not yet supported. Orders processed through the app won't display this section). You can use the testing site https://indiemelon.mystagingwebsite.com/ if you prefer, that already has processed orders.
- In the app, go to `Orders` > Tap in the order you just processed (or look for orders `2478` or `2572` in https://indiemelon.mystagingwebsite.com/ )
- When loaded, scroll down and observe the Subscriptions section renders correctly.

PD: I'm not a big fan of the amount of separator lines when there's multiple subscriptions (see order `2572`). I'll followup with design and iterate further if needed:

<img width="385" alt="Screenshot 2024-04-24 at 16 15 41" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/cf82b05e-c761-435f-8258-bb4340bdcfa0">
